### PR TITLE
[Compute] Automapper fix for bug in Update-AzDisk and Update-AzSnapshot 

### PR DESCRIPTION
**Pull request by haagha**
## Description

Bug in Update-AzDisk and Update-AzSnapshot causes null Tags property to be mapped as empty dictionaries. When passed to the PATCH call, it results in clearing user's Tags.

This bug fix will add a condition for the mapping to only map when Tag's property is not null.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by isra-fel** [comment]
+ @NoriZC + @wyunchi-ms  to review as well.
BTW @haagha please update the changelog.

**Comment by wyunchi-ms** [comment]
Is there any other cmdlets act like this way?

File: src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs
Comment (Sandido) [inline src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs:112]: I'm a bit confused as to why this will fix the issue. The issue is that customers have Tag values that are being lost, so their Tags are filled and are being mapped to an empty tag. 
Now this condition checks if the customer Tag has values in it and maps it only if there are tag values? Isn't that the same behavior as before?

File: src/Compute/Compute/ChangeLog.md
Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:29]: Add a test for this scenario.

**Review by Sandido — DISMISSED** [review]

**Comment by haagha** [comment]
@wyunchi-ms Thankfully we don't use Automapper for any other Update Objects. These are the only two.

File: src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs
Comment (haagha) [reply src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs:112]: @Sandido the issue is that in a PATCH call you only send items that need to be updated. If the tags are null, changing them to empty will be interpreted as Clearing out all tags. These conditions check to not map tags if they are null.

**Comment by haagha** [comment]
Adding a few other changes requested by CXP before merging. Please hold on merge, Thanks.

File: src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs
Comment (vidai-msft) [reply src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs:117]: @haagha if the end users do want to clear the tags, would they fail to achieve that?

Comment (haagha) [reply src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs:117]: @vidai-msft The users can clear tags like normally before by passing an empty dictionary. An empty dictionary would not meet the condition.

Comment (vidai-msft) [reply src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs:117]: @haagha OK. got it. The expected behavior is empty dictionary will clear the tag while null should do nothing.

**Review by vidai-msft — APPROVED** [review]

**Comment by vidai-msft** [comment]
@Sandido can you please approve this PR if you don't have any other question or concern?

**Comment by vidai-msft** [comment]
Gently ping @Sandido

**Comment by haagha** [comment]
@vidai-msft I've explained to @Sandido the behavior that will be because of this change. Please merge this.

**Pull request merged**
